CMakeLists.txt

```
@@ -35,6 +35,7 @@ endif()
 option(SOLC_LINK_STATIC "Link solc executable statically on supported platforms" OFF)
 option(SOLC_STATIC_STDLIBS "Link solc against static versions of libgcc and libstdc++ on supported platforms" OFF)
 option(STRICT_Z3_VERSION "Use the latest version of Z3" ON)
+option(DISABLE_WARNINGS "Disable all warning flags, treating warnings as errors and all other pedantic build flags" OFF) 
```

Technically this does not disable all warnings, just does not enable the extra warnings we normally want. What do you think about calling it `PEDANTIC` instead?

```suggestion
option(PEDANTIC "Enable extra warnings and pedantic build flags. Treat all warnings as errors." ON)
```